cda-core/src/diag_kernel/schema.rs

```diff
@@ -0,0 +1,361 @@
+            continue;
+        };
+        let Some(name) = STRINGS.get(param.short_name) else {
+            tracing::trace!("Mapping {ctx}: Parameter short name not found in strings. skipping");
+            continue;
+        };
+        let val = if let datatypes::ParameterValue::MatchingRequestParam(matching) = &param.value {
+            let Some(request_id) = request_id else {
+                tracing::trace!(
+                    "Mapping {ctx}: Parameter is a MatchingRequestParam within a request context."
+                );
+                continue;
+            };
+            let Some(val) = ecu_db
+                .requests
+                .get(&request_id)
+                .and_then(|req| {
+                    req.params.iter().find_map(|p| {
+                        ecu_db.params.get(p).and_then(|p| {
+                            // note: check explicitly if its safe to convert it to a u32
+                            // if it is < 0 we can safely assume that no param will match
+                            if matching.request_byte_pos > 0
+                                && (p.byte_pos == matching.request_byte_pos as u32)
+                            {
+                                Some(p)
+                            } else {
+                                None
+                            }
+                        })
+                    })
+                })
+                .and_then(|matching_param| {
+                    let datatypes::ParameterValue::Value(val) = &matching_param.value else {
+                        return None;
+                    };
+                    Some(val)
+                })
+            else {
+                tracing::trace!(
+                    "Mapping {ctx}: Matching request parameter not found in request. skipping"
+                );
+                continue;
+            };
+            val
+        } else {
+            let datatypes::ParameterValue::Value(val) = &param.value else {
+                tracing::trace!("Mapping {ctx}: Parameter is not a value. skipping");
+                continue;
+            };
+            val
+        };
+        let Some(dop) = ecu_db.data_operations.get(&val.dop) else {
+            tracing::trace!("Mapping {ctx}: DOP not found in ECU database. skipping");
+            continue;
+        };
+        let default_value = match val.default_value {
+            Some(ref_) => STRINGS.get(ref_),
+            None => None,
+        }
+        .unwrap_or_default();
+
+        let schema = match schema {
+            Some(ref mut s) => s,
+            None => {
+                schema = Some(schemars::json_schema!(true));
+                schema.as_mut().unwrap()
+            }
+        };
+
+        match &dop.variant {
+            datatypes::DataOperationVariant::Normal(normal_dop) => {
+                // todo: schould we add a description or something
```

Comment:
question: Do we need this todo and if so, should we create an issue for it, so it doesn't get lost? 
In my opinion if we document how DOPs work we should do this on the data operation itself and not within the schema.

Reply:
with that argumentation even keeping it works for 👍